### PR TITLE
Allow quantile function to accept tuples

### LINKED ISSUE #18458: quantile! should accept Tuple{Vararg{Real}} for quantile probabilities
At the moment it accepts only a single Real or an AbstractArray.

### PR DESCRIPTION
**Pull request by simonster**
Fixes #18458

### CODE REVIEW

File: base/statistics.jl [hunk 1]
```diff
@@ -630,19 +630,10 @@ function quantile!(q::AbstractArray, v::AbstractVector, p::AbstractArray;
     if size(p) != size(q)
         throw(DimensionMismatch("size of p, $(size(p)), must equal size of q, $(size(q))"))
     end
+    isempty(q) && return q
 
-    isempty(v) && throw(ArgumentError("empty data vector"))
```
Comment (ararslan) [inline base/statistics.jl]: Seems like this should be preserved?

Comment (simonster) [inline base/statistics.jl]: It's still there. I just split out the previously duplicated code to the `_quantilesort` function.

Comment (ararslan) [inline base/statistics.jl]: Oh, I see now. Thanks and sorry for the noise.

File: base/statistics.jl [hunk 2]
```diff
@@ -605,10 +605,11 @@ median{T}(v::AbstractArray{T}, region) = mapslices(median!, v, region)
 """
     quantile!([q, ] v, p; sorted=false)
 
-Compute the quantile(s) of a vector `v` at the probabilities `p`, with optional output into
-array `q` (if not provided, a new output array is created). The keyword argument `sorted`
-indicates whether `v` can be assumed to be sorted; if `false` (the default), then the
-elements of `v` may be partially sorted.
+Compute the quantile(s) of a vector `v` at the probability or vector or tuple of
```
Comment (ararslan) [inline base/statistics.jl]: "probability or vector or tuple" sounds kind of confusing IMO. Would something like "at the probability or probabilities `p`, given as a single value, a vector, or a tuple" be reasonable?

Comment (simonster) [inline base/statistics.jl]: Sure. Feel free to push to the PR.

Comment (ararslan) [inline base/statistics.jl]: Okay, will do. Just wanted to make sure you thought that wording was reasonable.

**Pull request merged**